folium/features.py

```diff
@@ -413,22 +413,23 @@ class GeoJson(Layer):
                 }
             });
         };
-        var {{ this.get_name() }} = L.geoJson(null, {
-            {%- if this.smooth_factor is not none  %}
-                smoothFactor: {{ this.smooth_factor|tojson }},
-            {%- endif %}
-                onEachFeature: {{ this.get_name() }}_onEachFeature,
-            {% if this.style %}
-                style: {{ this.get_name() }}_styler,
-            {%- endif %}
-        }).addTo({{ this._parent.get_name() }});
+        var options = {
+        {%- if this.smooth_factor is not none  %}
+            smoothFactor: {{ this.smooth_factor|tojson }},
+        {%- endif %}
+            onEachFeature: {{ this.get_name() }}_onEachFeature,
+        {% if this.style %}
+            style: {{ this.get_name() }}_styler,
+        {%- endif %} };
         {%- if this.embed %}
-            {{ this.get_name() }}.addData({{ this.data|tojson }});
+        var {{ this.get_name() }} = L.geoJson(
+            {{ this.data|tojson }}, options).addTo({{ this._parent.get_name() }});
         {%- else %}
-            $.ajax({url: {{ this.embed_link|tojson }}, dataType: 'json', async: true,
-                success: function(data) {
-                    {{ this.get_name() }}.addData(data);
-            }});
+        var json_url = $.ajax({url: {{ this.embed_link|tojson }}, dataType: 'json', async: true })
+        $.when(json_url).done(function() {
```

Comment:
You can link this to the ajax call directly: `$.ajax().done(function() {});`

Reply:
Have you tried? Because the problem I had was related with async ajax request and because of that I used a variable to check if the request was finished. I don't have experience with JS. I just searched about it and I found that solution in a blog post.

Reply:
It seems to work [here](https://github.com/python-visualization/folium/pull/1190/files#diff-188d69782eb76f735a9d6a3c508f3aa3R432), and is also listed in the examples in the jQuery documentation, so assume it should be fine.